### PR TITLE
View article page

### LINKED ISSUE #23: View article page
**Description of feature/enhancement**
Create the view article page. Note the differences between the different types of articles (draft, published, archived)
Screen Shot 2022-12-11 at 10 43 23 PM
Screen Shot 2022-12-11 at 10 43 16 PM
Screen Shot 2022-12-11 at 10 43 10 PM

_insert link to figma etc_

### PR DESCRIPTION
**Pull request by CeereeC**
Closes #23 

**Overview of changes:**
- Added Skeleton of View Article Page

**Anything you'd like to highlight/discuss:**
- No functionality yet
- Currently View Articles Page exists as a standalone page, with its own route. Not sure if this is the correct behaviour.
- If the correct behaviour is for the View Articles Page to be a redirect from the View Articles List Page, I can make the change in Issue #29 when I link these 2 pages together
- Intending to add a bit of regex to check that the route has the articlename-id structure

### CODE REVIEW

File: components/ViewArticlePage/ArticlePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      >
+        <Box
+          component="img"
+          src={EditIcon.src}
+          sx={{
+            display: 'flex',
+            justifyContent: 'center',
+            height: 32,
+          }}
+        />
```
Comment (wilinetan) [inline components/ViewArticlePage/ArticlePanel.tsx:39]: I was wondering why are we not able to just use the edit icon provided by mui? https://mui.com/material-ui/material-icons/?query=edit&selected=Edit

File: components/ViewArticlePage/ViewsPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { Paper, Typography } from '@mui/material'
+
+const ViewsPanel = ({ viewCount }: { viewCount: number }) => {
```
Comment (wilinetan) [inline components/ViewArticlePage/ViewsPanel.tsx:3]: this component is the same as RevisionsPanel. so we can just abstract it out.
```
import { Paper, Typography } from '@mui/material'

enum CountType {
  REVISION = 'Revision',
  VIEWS = 'Views'
}

const ArticleCountPanel = ({ count, countType }: { count: number, countType: CountType }) => {
  return (
    <Paper
      elevation={2}
      sx={{
        height: 165,
        borderRadius: 3,
        marginBottom: '20px',
        padding: '12px 24px',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-evenly',
      }}
    >
      <Typography fontSize={'21px'} fontWeight={700}>
        `Number of ${countType}:`
      </Typography>
      <Typography textAlign={'center'} fontSize={'28px'} fontWeight={400}>
        {count}
      </Typography>
      <Typography
        textAlign={'right'}
        color="#009DD7"
        sx={{ textDecoration: 'underline' }}
      >
        `View ${countType}&apos;s History`
      </Typography>
    </Paper>
  )
}

export default ArticleCountPanel
```

then when using it:
```
<ArticleCountPanel count={13} countType={CountType.REVISION} />
<ArticleCountPanel count={100} countType={CountType.VIEWS} />
```

File: components/ViewArticlePage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import ArticlePanel from './ArticlePanel'
+import ButtonsPanel from './Buttons/ButtonsPanel'
+import DetailsPanel from './DetailsPanel'
+import RevisionsPanel from './RevisionsPanel'
+import ViewsPanel from './ViewsPanel'
+
+const ViewArticlePage = () => {
+  // Some code to ping backend for article data
+
+  const [articleType, setArticleType] = useState<string>('Published')
```
Comment (wilinetan) [inline components/ViewArticlePage/index.tsx:12]: can use the same enum that was used for donations. `CampaignStatus`

File: components/ViewArticlePage/Buttons/ButtonsPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    <>
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'space-evenly',
+          textTransform: 'uppercase',
+        }}
+      >
+        <DeleteButton />
+        {articleType == 'Published' ? <ArchiveButton /> : <PublishButton />}
```
Comment (wilinetan) [inline components/ViewArticlePage/Buttons/ButtonsPanel.tsx:17]: use the enum

File: components/ViewArticlePage/DetailsPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    <Paper
+      elevation={2}
+      sx={{
+        height: 410,
+        borderRadius: 3,
+        marginBottom: '20px',
+        padding: '24px',
+      }}
+    >
+      <Typography fontSize={'21px'} fontWeight={700} marginBottom="10px">
```
Comment (wilinetan) [inline components/ViewArticlePage/DetailsPanel.tsx:27]: just a nit remark, when passing strings, we dont need to wrap it in a `{}`. so this line could just be `fontSize="12px"`. but u don't have to change it, just to let u know

**Review by wilinetan — COMMENTED** [review]
looks good!

**Comment by wilinetan** [comment]
```
If the correct behaviour is for the View Articles Page to be a redirect from the View Articles List Page, I can make the change in Issue https://github.com/dsc-project-mr/mr-fe-internal/issues/29 when I link these 2 pages together
```
> yes it will be a redirect from the View Articles List Page. what kind of changes are you thinking of doing?

File: components/ViewArticlePage/ArticlePanel.tsx (hunk 1 above)
Comment (CeereeC) [reply components/ViewArticlePage/ArticlePanel.tsx:39]: Screenshot 2022-12-20 at 12 13 30 AM

I just followed the image in the Figma, seems to be a custom vector image the UI team used.

**Review by wilinetan — APPROVED** [review]
looks good! for the page url path, i think in the sidebar component for articles it is `/content/articles`? so for this it can be at `/content/articles/[articlename_id]`

**Comment by CeereeC** [comment]
> looks good! for the page url path, i think in the sidebar component for articles it is `/content/articles`? so for this it can be at `/content/articles/[articlename_id]`

Alright got it!

**Comment by wilinetan** [comment]
once you have updated the url paths, can merge already

**Pull request merged**